File: 392-is-subsequence/392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        j, n = 0, len(s)
        for i in t:
            if not s: return True
            j = j + 1 if j < n and i == s[j] else j
            
        return j == n

File: 392-is-subsequence/392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int idx = 0, n = s.size(), m = t.size();
        if (m < n) return false;
        
        for (int i = 0; i < m; i++) {
            if (s[idx] == t[i])
                idx++;
        }
        return idx == n;
    }
};